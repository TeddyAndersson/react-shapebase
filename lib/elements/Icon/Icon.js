import React from 'react';
import _ from 'lodash';
import PropTypes from 'prop-types';
import styled, {css} from 'styled-components';
import fontawesome from '@fortawesome/fontawesome'
import FontAwesomeIcon from '@fortawesome/react-fontawesome'
import solids from '@fortawesome/fontawesome-free-solid'

fontawesome.library.add(solids)

const Element = props => {
    const {
        className,
        name,
        color,
        size,
    } = props;

    return (
        <FontAwesomeIcon className={className} icon={name} />
    );
}

Element.propTypes = {

    /**
     * Additonal classes.
     */
    className: PropTypes.string,

    /**
     * The name of the fontawesome icon.
     */
    name: PropTypes.string.isRequired,

    /**
     * The the color of the icon.
     */
    color: PropTypes.string,
    /**
     * The the size of the icon.
     */
    size: PropTypes.number,
}

Element.defaultProps = {
    color: "#000",
    size: 1,
}

/** Styling the Component with styled-components */
const Icon = styled(Element)`
    color: ${props => props.color};
    font-size: ${props => props.size ? props.size : 2}em;
`;

export default Icon;