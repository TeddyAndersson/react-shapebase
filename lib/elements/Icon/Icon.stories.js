import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Grid from '../Grid';
import Icon from './Icon';

const lorem = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."

storiesOf('Icon', module)
  .add('Icon', () => (
    <Grid columns={1} rowGap={1}>
        <Icon name="rocket" />
        <Icon name="smile" />
        <Icon name="check" size={1} color={"tomato"} />
    </Grid>
  ));