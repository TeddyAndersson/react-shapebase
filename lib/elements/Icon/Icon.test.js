import React from 'react';
import Icon from './Icon'
import renderer from 'react-test-renderer';

test('should render Icon', () => {
  const component = renderer.create(
    <Icon name="rocket"/>
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
})
