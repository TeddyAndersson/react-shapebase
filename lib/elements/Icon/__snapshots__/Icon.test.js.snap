// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Icon 1`] = `
<span
  aria-hidden={true}
  className="fa fa-rocket sc-bdVaJa dSDeol"
/>
`;
